File: day05/maze.py
#!/usr/bin/env python3
from sys import argv

# read in the program
filename = argv[1]
pgm = []
with open(filename) as f:
    for line in f:
        pgm.append(int(line))

# run the program
ip = 0
steps = 0
while ip >= 0 and ip < len(pgm):
    tmp = ip
    ip += pgm[tmp]
    pgm[tmp] += 1
    steps += 1

print("result1:", steps)

File: day05/maze.pl
#!/usr/bin/env perl
use strict;
use warnings;
use feature qw(:5.24);
use experimental qw(signatures);

# read in the program
my @pgm;

while (<>) {
    chomp;
    push @pgm, $_;
}

# run the program;
my $ip = 0;
my $steps = 0;
while ($ip >= 0 && $ip <= $#pgm) {
    my $tmp = $ip;
    $ip += $pgm[$tmp];
    $pgm[$tmp]++;
    $steps++;
}

say "result1: $steps";
